OptimizelySDK.Tests/NotificationTests/NotificationCenterTests.cs

```diff
@@ -0,0 +1,195 @@
+
+            // Add track notification.
+            NotificationCenter.AddNotification(notificationTypeTrack, TestNotificationCallbacks.TestTrackCallback);
+
+            // Verify that callbacks added successfully.
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that ClearAllNotifications remove all the callbacks.
+            NotificationCenter.ClearAllNotifications();
+            Assert.AreEqual(0, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestFireNotifications()
+        {
+            var decisionNotification = NotificationCenter.NotificationType.Decision;
+            var trackNotification = NotificationCenter.NotificationType.Track;
+            var config = ProjectConfig.Create(TestData.Datafile, LoggerMock.Object, new NoOpErrorHandler());
+
+            // Mocking notification callbacks.
+            var notificationCallbackMock = new Mock<TestNotificationCallbacks>();
+            notificationCallbackMock.Setup(nc => nc.TestDecisionCallback(It.IsAny<Experiment>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()));
+            notificationCallbackMock.Setup(nc => nc.TestAnotherDecisionCallback(It.IsAny<Experiment>(), 
+                It.IsAny<string>(), It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()));
+
+            // Adding decision notifications.
+            NotificationCenter.AddNotification(decisionNotification, notificationCallbackMock.Object.TestDecisionCallback);
+            NotificationCenter.AddNotification(decisionNotification, notificationCallbackMock.Object.TestAnotherDecisionCallback);
+
+            // Adding track notifications.
+            NotificationCenter.AddNotification(trackNotification, notificationCallbackMock.Object.TestTrackCallback);
+            
+            // Firing decision type notifications.
+            NotificationCenter.FireNotifications(decisionNotification, config.GetExperimentFromKey("test_experiment"), "testUser",
+                new UserAttributes(), config.GetVariationFromId("test_experiment", "7722370027"), null);
+
+            // Verify that only the registered notifications of decision type are called.
+            notificationCallbackMock.Verify(nc => nc.TestDecisionCallback(It.IsAny<Experiment>(), It.IsAny<string>(), 
+                It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()), Times.Exactly(1));
+            notificationCallbackMock.Verify(nc => nc.TestAnotherDecisionCallback(It.IsAny<Experiment>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()), Times.Exactly(1));
+            notificationCallbackMock.Verify(nc => nc.TestTrackCallback(It.IsAny<string>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<EventTags>(), It.IsAny<LogEvent>()), Times.Never);
```

Comment:
It would be great if you can add logs as well.

Reply:
I don't think I am getting you. We can't add logs in mock functions.

Reply:
method main actual aur yiehan verify karo. :P